### DISCUSSION TITLE
Intermittent ERROR 500 - INTERNAL SERVER ERROR

### DISCUSSION THREAD

**Discussion by ghubmlafontaine** [kimai/kimai#3343] [General]
**Error 500 - internal server error**
I tried Kimai earlier version a couple year ago to evaluate it.   Last month, I decided to actually try it for real.   I am on a shared hosting service (Hostinger, not to name it).  I performed the upgrade to KImai2, going to version 1.19. and created a new DB for Kimai2.   I experienced the problem described below, so I went ahead and upgraded to 1.20.2  I am still have the same problem:

I can connect OK
I can browse some but not all of the sub-menu of the app  for example:
Dashboard  OK                                My time Error 500
Weekly working hours Error 500     Calendar Error 500
Export OK                                        Timesheet OK
Reporting Error 500                         Doctor  OK

Can I update the Database?  yes and no
Create a new user    OK
Create a new team  Error 500

Finally  I ran a chmod _R 755 kimai  to make sure file security wasn't a problem, with no improvement
 

**Logfile**
```
I flushed the log file at the 1.19 upgrade after experiencing the problem to start fresh.  I noticed that nothing gets log and it remains desperately empty until now...
```

**Additional context**
 - Apache v 2.4.53
 - Kimai version: 1.20.2
 - PHP version: 7.3  ( applied to Apache)
 - PHP default version on the server is 8.0
 - MySQL (10.5.15-MariaDB-cll-lve) 

The software seems to work OK so I suspect the problem lies with my implementation; Bad link, missing routing, library or something, I am just not sure where to go, which log to parse to get a clue to the end of the issue?
How can I know ensure and validate that my new Kimai2 DB was really updated appropriately?

**Comment by kevinpapst** [comment]
If there is no log, nobody can really help you.
Try to fix permissions first. 
Then validate that you are not installing composer dependencies with PHP 8 and use Kimai with 7.3 in the browserr.

**Reply by ghubmlafontaine** [reply]
Hi,
Is it safe to rerun the update step by step using php73?
I see I can activate all logging for PHP which I will do.  Were else should I look for logs?

**Reply by ghubmlafontaine** [reply]
Bingo!    

I re-ran the install following the instructions to update to 1.20.2.  this time I forced the command line with php73 and also went back to apache and assign it to  PHP73.  

I also re run the file permission command recommended
 .... for Debian/Ubuntu (to be executed inside the Kimai directory):

chmod -R g+r .
chmod -R g+rw var/
chmod -R g+rw public/avatars/

et voila!    all menu screen are now showing.....   

I look forward learning Kimai ! 

Many Thanks for the hint 👍

**Reply by kevinpapst** [reply]
Glad to read that you solved the mystery so quick 😄
